experimental/dds/tree2/src/core/rebase/utils.ts

```diff
@@ -56,10 +56,13 @@ export interface RebasedCommits<TChange> {
  *
  * The source and target branch must share an ancestor.
  * @param changeRebaser - the change rebaser responsible for rebasing the changes in the commits of each branch
+ * @param computeSourceChange - if `true` the returned tuple will include the net change to the source branch.
```

Comment:
Instead of passing a flag like this, I wonder if it would be better to always return a thunk that can generate the net source change. That thunk could even be included in the `afterChange` event so that we can avoid computing the net source change when we have subscribers to the event that don't actually consume the net source change.

This might end up being clearer once we add the code path that rebases over composition because once we do that, opting into that code path will always generate the net source changes, which would make for a confusing contract: "we don't compute it unless you ask for it, or unless you happen to choose the rebasing method that is based on composition".

I won't make the change until I get some buy in from at least one more person.

Reply:
I like this idea. It also makes the return value more uniform.

Reply:
I was going to suggest doing a thunk before I even found this comment, so it's a yes from me as well :D